runtime/near-vm-runner/src/near_vm_runner.rs

```diff
@@ -258,7 +258,7 @@ impl NearVM {
                 // that particular function call will be slower. Not to mention there isn't a
                 // strong guarantee on the upper bound of the memory that the contract runtime may
                 // require.
-                LimitedMemoryPool::new(8, 64 * 1024 * 1024).unwrap_or_else(|e| {
+                LimitedMemoryPool::new(256, 1 * 1024 * 1024).unwrap_or_else(|e| {
```

Comment:
The size of each individual memory in the pool grows as required. This grow operation is fallible, unfortunately, but at the very least with the change to the LIFO queue for the memory pool, in most "regular" applications only the required number of memories should actually grow. For a validator that tracks 4 shards, that would mean just 4 of these memories.